### PR TITLE
test database management: creation, deletion, index

### PR DESCRIPTION
**Pull request by cendhu**
Signed-off-by: senthil <cendhu@gmail.com>

### CODE REVIEW

File: test/setup/server_setup.go [hunk 1]
```diff
@@ -220,6 +220,26 @@ func (s *Server) GetDBStatus(t *testing.T, dbName string) (*types.GetDBStatusRes
...
+	query := &types.GetDBIndexQuery{
+		UserId: s.AdminID(),
+		DbName: dbName,
+	}
+	response, err := client.GetDBIndex(
+		&types.GetDBIndexQueryEnvelope{
+			Payload:   query,
+			Signature: testutils.SignatureFromQuery(t, s.AdminSigner(), query),
+		},
+	)
```
[the remark this answers is not in the document]
Comment (cendhu) [reply test/setup/server_setup.go:238]: Other users can query the index.

File: test/db/db_index_management_test.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+			require.True(t, response.GetResponse().Exist)
+
+			index, err := s.GetDBIndex(t, dbName)
+			require.NoError(t, err)
+			require.Empty(t, index.GetResponse().GetIndex())
+		}
+	})
+
+	// update database db1 with an index
+	t.Run("update db1 with an index", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cendhu) [reply test/db/db_index_management_test.go:85]: There is a bug in the existing code that allows this transaction to pass. Need to fix the bug first.

File: test/db/db_index_management_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+		index, err := s.GetDBIndex(t, "db3")
+		require.NoError(t, err)
+
+		expectedIndex, err := json.Marshal(db3Index.AttributeAndType)
+		require.NoError(t, err)
+		require.Equal(t, string(expectedIndex), index.GetResponse().GetIndex())
+	})
+
+	t.Run("delete db1 and db3", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cendhu) [reply test/db/db_index_management_test.go:127]: Adding a flag does require significant code changes at the sdk, server, and test code. Needs to do it in a separate PR. For now, we can test the default test behaviour.

File: test/setup/server_setup.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cendhu) [reply test/setup/server_setup.go]: Done

File: test/db/db_index_management_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cendhu) [reply test/db/db_index_management_test.go]: Done

File: test/db/db_index_management_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	})
+	require.NoError(t, err)
+	require.NotNil(t, receipt)
+
+	for _, user := range users {
+		respEnv, err := s.QueryUser(t, user.GetUser().GetId())
+		require.NoError(t, err)
+		require.Equal(t, user.User, respEnv.GetResponse().User)
+	}
+}
```
[the remark this answers is not in the document]
Comment (cendhu) [reply test/db/db_index_management_test.go]: Done

File: test/db/db_index_management_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cendhu) [reply test/db/db_index_management_test.go:159]: Done

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**